scripts/dualtor_neighbor_check.py

```diff
@@ -0,0 +1,427 @@
+    )
+    args = parser.parse_args()
+
+    if args.log_output == LogOutput.STDOUT:
+        if args.log_level is None:
+            args.log_level = logging.WARNING
+        else:
+            args.log_level = logging.getLevelName(args.log_level)
+
+        if args.syslog_level is not None:
+            parser.error("Received syslog level with log output to stdout.")
+    if args.log_output == LogOutput.SYSLOG:
+        if args.syslog_level is None:
+            args.syslog_level = SyslogLevel.NOTICE
+        else:
+            args.syslog_level = SyslogLevel[args.syslog_level]
+
+        if args.log_level is not None:
+            parser.error("Received stdout log level with log output to syslog.")
+
+    return args
+
+
+def write_syslog(level, message, *args):
+    if args:
+        message %= args
+    if level == SyslogLevel.ERROR:
+        syslog.syslog(syslog.LOG_ERR, message)
+    elif level == SyslogLevel.NOTICE:
+        syslog.syslog(syslog.LOG_NOTICE, message)
+    elif level == SyslogLevel.INFO:
+        syslog.syslog(syslog.LOG_INFO, message)
+    elif level == SyslogLevel.DEBUG:
+        syslog.syslog(syslog.LOG_DEBUG, message)
+    else:
+        syslog.syslog(syslog.LOG_DEBUG, message)
+
+
+def config_logging(args):
+    """Configures logging based on arguments."""
+    global SYSLOG_LEVEL
+    global WRITE_LOG_ERROR
+    global WRITE_LOG_WARN
+    global WRITE_LOG_INFO
+    global WRITE_LOG_DEBUG
+    if args.log_output == LogOutput.STDOUT:
+        logging.basicConfig(
+            stream=sys.stdout,
+            level=args.log_level,
+            format="%(message)s"
+        )
+        WRITE_LOG_ERROR = logging.error
+        WRITE_LOG_WARN = logging.warning
+        WRITE_LOG_INFO = logging.info
+        WRITE_LOG_DEBUG = logging.debug
+    elif args.log_output == LogOutput.SYSLOG:
+        SYSLOG_LEVEL = args.syslog_level
+        WRITE_LOG_ERROR = functools.partial(write_syslog, SyslogLevel.ERROR)
+        WRITE_LOG_WARN = functools.partial(write_syslog, SyslogLevel.NOTICE)
+        WRITE_LOG_INFO = functools.partial(write_syslog, SyslogLevel.INFO)
+        WRITE_LOG_DEBUG = functools.partial(write_syslog, SyslogLevel.DEBUG)
+
+
+def run_command(cmd):
+    """Runs a command and returns its output."""
+    WRITE_LOG_DEBUG("Running command: %s", cmd)
+    try:
+        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        (output, _) = p.communicate()
+    except Exception as details:
+        raise RuntimeError("Failed to run command: %s", details)
+    WRITE_LOG_DEBUG("Command output: %s", output)
+    WRITE_LOG_DEBUG("Command return code: %s", p.returncode)
+    if p.returncode != 0:
+        raise RuntimeError("Command failed with return code %s: %s" % (p.returncode, output))
+    return output.decode()
+    
+
+def read_tables_from_db():
+    """Reads required tables from db."""
+    load_cmd = "sudo redis-cli SCRIPT LOAD \"%s\"" % DB_READ_SCRIPT
```

Comment:
We are loading the same lua script at each run. Will it consume more and more memory? Or the previous one is overwriten?

Reply:
Add support to cache the lua sha1 into a temporary key in `APPL_DB`.